flask/app/analyses.py

```diff
@@ -178,6 +178,38 @@ def create_analysis():
     except:  # LOGIN DISABLED
         requester_id = updated_by_id = 1
 
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+    else:
+        user_id = current_user.user_id
```

Comment:
You should move this part up earlier so the preceding `requester_id` assignment can make use of this assumed user identity as well.